pulp/apis/sas_api.py

```diff
@@ -0,0 +1,623 @@
+    "ABORTED": constants.LpStatusNotSolved,
+    "ABORT_SOL": constants.LpStatusNotSolved,
+    "ABORT_NOSOL": constants.LpStatusNotSolved,
+    "OUTMEM_SOL": constants.LpStatusNotSolved,
+    "OUTMEM_NOSOL": constants.LpStatusNotSolved,
+    "FAILED": constants.LpStatusNotSolved,
+    "FAIL_SOL": constants.LpStatusNotSolved,
+    "FAIL_NOSOL": constants.LpStatusNotSolved,
+    "ERROR": constants.LpStatusNotSolved,
+}
+
+CAS_OPTION_NAMES = [
+    "hostname",
+    "port",
+    "username",
+    "password",
+    "session",
+    "locale",
+    "name",
+    "nworkers",
+    "authinfo",
+    "protocol",
+    "path",
+    "ssl_ca_list",
+    "authcode",
+]
+
+
+class SAS94(LpSolver):
+    name = "SAS94"
+
+    try:
+        global saspy
+        import saspy
+
+    except ImportError:
+
+        def available(self):
+            """True if SAS94 is available."""
+            return False
+
+        def actualSolve(self, lp, callback=None):
+            """Solves a well-formulated lp problem."""
+            raise PulpSolverError("SAS94 : Not Available")
+
+    else:
+        saspy.logger.setLevel(log.level)
+        def __init__(
+            self,
+            mip=True,
+            msg=True,
+            warmStart=False,
+            **solverParams,
+        ):
+            """
+            :param bool mip: if False, assume LP even if integer variables
+            :param bool msg: if False, no log is shown
+            :param bool warmStart: if False, no warmstart or initial primal solution provided.
+            """
+            LpSolver.__init__(
+                self,
+                mip=mip,
+                msg=msg,
+            )
+            self._sas = saspy.SASsession()
+            self.warmStart=warmStart
+            # Only named options are allowed in SAS solvers.
+            self.solverOptions = {key:value for key, value in solverParams.items()}
+        def available(self):
+            """True if SAS94 is available."""
+            return True
+
+        def _create_statement_str(self, statement):
+            """Helper function to create the strings for the statements of the proc OPTLP/OPTMILP code."""
+            stmt = self.solverOptions.pop(statement, None)
+            if stmt:
+                return (
+                    statement.strip()
+                    + " "
+                    + " ".join(option + "=" + str(value) for option, value in stmt.items())
+                    + ";"
+                )
+            else:
+                return ""
+
+        def toDict(self):
+            data = dict(solver=self.name)
+            for k in ["mip", "msg", "warmStart"]:
+                try:
+                    data[k] = getattr(self, k)
+                except AttributeError:
+                    pass
+            data.update(self.solverOptions)
+            return data
+
+        def _create_tmp_files(self, name, *args):
+
+            return (f"{name}-pulp.{n}" for n in args)
+
+        def _delete_tmp_files(self, *args):
```

Comment:
there already exists a `delete_tmp_files` method that does almost exactly this. No need to create a new one. The same goes for `_create_tmp_files`